lib/common/model.js

```diff
@@ -220,12 +232,18 @@ function ModelFactory (
                 if(options && options.sort) {
                     cursor = cursor.sort(options.sort);
                 }
-                return cursor.toArray();
+
+                return Promise.resolve(cursor.toArray()).then(function(docs){
+                    return _.map(docs, self.deSerialize);
+                });
```

Comment:
cursor.toArray() may be ```null```
```Promise.map(null, ....)``` will be rejected.
And also ```cursor.toArray()``` only can be executed only **once** and then cursor will be exhausted.
So also can't use 
``` 
if(cursor.toArray()){
    Promise.map(cursor.toArray, self.deSerialize) 
}
```